src/js/tech/tech.js

```diff
@@ -851,13 +851,23 @@ Tech.withSourceHandlers = function(_Tech) {
     return this;
   };
 
+  let previousSource;
+
   // On the first loadstart after setSource
   _Tech.prototype.firstLoadStartListener_ = function() {
+    previousSource = this.currentSource_;
     this.one(this.el_, 'loadstart', _Tech.prototype.successiveLoadStartListener_);
   };
 
   // On successive loadstarts when setSource has not been called again
   _Tech.prototype.successiveLoadStartListener_ = function() {
+
+    // if the second loadstart is for the same source
+    // it is an invalid loadstart, set this event to trigger again
+    if (previousSource.src === this.currentSource_.src) {
```

Comment:
Wouldn't we just want to compare `this.currentSource_` with `this.el_.currentSrc`?